client/src/unifycr.c

```diff
@@ -1931,7 +1931,10 @@ static int unifycr_init(int rank)
         if (env) {
             strcpy(external_data_dir, env);
         } else {
-            strcpy(external_data_dir, EXTERNAL_DATA_DIR);
+            fprintf(stderr, "UNIFYCR_EXTERNAL_DATA_DIR not set to an SSD path "
+                            "(i.e UNIFYCR_EXTERNAL_DATA_DIR=/mount/ssd):%s : %d\n",
+                            __FILE__, __LINE__);
+            exit(UNIFYCR_FAILURE);
```

Comment:
Rather than calling exit() here, what would you think about propagating an error back so unifycr_mount() fails? Shared libraries shouldn't normally call exit().

Reply:
@nedbass maybe returning UNIFYCR_FAILURE? It looks like that is what the other calls are doing in unifycrfs_mount?

Reply:
Yes, I agree.

Longer term I think we should design a more robust way to return error information to the application, since there's a variety of reasons why `unifycr_mount()` might fail. A simple option might be to set errno, and provide documentation on how to interpret the error codes. We might want to think about some auxiliary interface that could return diagnostic messages that the application could print.